model/validation.go

```diff
@@ -204,16 +204,367 @@ func (t Tags) Validate() error {
 	return errs
 }
 
+// Validate ensures tags are well-formed, but does not insist that tags exist
+func Validate(t map[string]string) error {
```

Comment:
Is there any reason why not treat empty tags and 0 tags as same? Tags are only used as lower-bounds, in which case seems to be the same thing. I'm mostly concerned with duplicating code, so feel free to change the original tags.validate() to fit your needs.